File: Dynamic Programming/python/45.py
class Solution:
    def jump(self, nums):
        jump = 0
        maxreached = 0
        reach = 0
        for i in range(len(nums) - 1):
            maxreached = max(maxreached, nums[i] + i)
            if i == reach:
                jump += 1
                reach = maxreached
        return jump

File: Dynamic Programming/java/45.java
class Solution {
    public int jump(int[] nums) {
        int jump = 0;
        int maxreached = 0;
        int reach = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            maxreached = Math.max(maxreached, nums[i] + i);
            if (i == reach) {
                jump++;
                reach = maxreached;
            }
        }
        return jump;
    }
}